internal/internal_workflow_testsuite.go

```diff
@@ -840,11 +842,15 @@ func (env *testWorkflowEnvironmentImpl) executeActivityWithRetryForTest(
 			backoff := getRetryBackoffWithNowTime(p, task.GetAttempt(), *request.Reason, env.Now(), expireTime)
 			if backoff > 0 {
 				// need a retry
-				task.Attempt = common.Int32Ptr(task.GetAttempt() + 1)
 				waitCh := make(chan struct{})
 				env.postCallback(func() { env.runningCount-- }, false)
 				env.registerDelayedCallback(func() {
 					env.runningCount++
+					task.Attempt = common.Int32Ptr(task.GetAttempt() + 1)
+					activityID := string(task.TaskToken)
```

Comment:
Does such task token work if the same activity is invoked from multiple child workflows?

Reply:
Good catch! It won't work, and it is a bug. :)